### ISSUE TITLE
AddBasePath appends `/` to routes

### ISSUE THREAD

**Issue by polds** [cardinalby/hureg#2]
I _think_ this is a Hureg issue because Huma doesn't appear to do this. I was creating a "custom method" endpoint per Google's [AIP-136](https://google.aip.dev/136) spec. Namely `/collection:customAction`. In a Huma Operation if I set the route to `/collection:customAction` it correctly sets that exact route. However if I use Hureg + AddBasePath and set `/collection` as the base path and then attempt to create an operation with a route `:customAction` I was surprised to see the spec be updated to be `/collection/:customAction`.

You can see it here:
```golang
api = api.AddBasePath("/collection")

op := huma.Operation{
   OperationID: "customAction",
   Method:      http.MethodGet,
   Path:        ":customAction", // Expected to live at `/collection:customAction`
}

hureg.Register(api, op, nil)
```

Is it possible to keep the intended `/collection:customAction` while still leveraging the `AddBasePath` method?

**Comment by cardinalby** [comment]
Hi! You are correct that it's hureg that adds path separators implicitly. 
As I know, with chi it's also not possible to achieve that since it just panics if you tries to add route not starting from "/" to a group, hureg kind of imitates this behavior but adds separator implicitly instead of panicing. 

There is no fundamental limitations why I can't change that. I believe the general idea why you are supposed to start your paths from "/" is that you can use any `api` (even without a base path) in this case.

What is the better approach in your opinion?  Just concat the strings without any separators?

P.S. Right now it's not possible via `AddBasePath`, but you can make an own operation handler similar to my implementation.

**Comment by polds** [comment]
I'm not sure which is better tbh. With `.AddBasePath("/collection")` I can successfully set a root operation with `""` to do things like `GET /collection`. If I remove the AddBasePath then Huma requires me to set `/collection` and `/collection:customAction`

Ideal case at least for me, considering I use `.AddBasePath` for groupings would be to continue to do so and be able to do things like:

```golang
api = api.AddBasePath("/collection")

get := huma.Operation{
   OperationID: "getCollection",
   Method:      http.MethodGet,
   Path:        "", // /collection
}

custom := huma.Operation{
  OperationID: "customAction",
  Method: http.MethodPost,
  Path: ":customAction", // /collection:customAction
}
```

But given I _can_ get away with it by not using `AddBasePath` and explicitly defining routes for `/collection` and `/collection:customAction` maybe it's not a huge deal for these groups to not use the AddBasePath? It just makes operation definition a little more verbose, but maybe that's fine.

Basically turning it into:

```golang
api = api.AddOpHandler(Op.AddTags("Collection"))

get := huma.Operation{
   OperationID: "getCollection",
   Method:      http.MethodGet,
   Path:        "/collection",
}

custom := huma.Operation{
  OperationID: "customAction",
  Method: http.MethodPost,
  Path: "/collection:customAction",
}
```
